src/ThriveWorkerUnit.sol

```diff
@@ -0,0 +1,286 @@
+    address public immutable rewardToken;
+    uint256 public immutable rewardAmount;
+    uint256 public immutable maxRewards;
+    uint256 public validationRewardAmount;
+    address public assignedAddress;
+    uint256 public maxCompletionsPerUser;
+    uint256 public deadline;
+    string public validationMetadata;
+    string public metadataVersion;
+    string public metadata;
+
+    EnumerableSet.AddressSet private validators;
+    EnumerableSet.Bytes32Set private requiredBadges;
+
+    mapping(address => uint256) public completions;
+
+    IBadgeQuery public badgeQuery;
+    bool public ready;
+
+    event Initialized();
+    event ConfirmationAdded(
+        address indexed contributor,
+        string validationMetadata,
+        uint256 rewardAmount,
+        address indexed validator
+    );
+    event Withdrawn(address token, uint256 amount);
+
+    modifier onlyModerator() {
+        require(
+            msg.sender == moderator,
+            "ThriveProtocol: only the moderator can perform this action"
+        );
+        _;
+    }
+
+    modifier onlyValidator() {
+        require(
+            validators.contains(msg.sender),
+            "ThriveProtocol: only a validator can perform this action"
+        );
+        _;
+    }
+
+    modifier onceReady() {
+        require(ready, "ThriveProtocol: contract is not ready");
+        _;
+    }
+
+    constructor(
+        address _moderator,
+        address _rewardToken,
+        uint256 _rewardAmount,
+        uint256 _maxRewards,
+        uint256 _deadline,
+        uint256 _maxCompletionsPerUser,
+        address[] memory _validators,
+        string memory _validationMetadata,
+        string memory _metadataVersion,
+        string memory _metadata,
+        address _badgeQuery
+    ) {
+        require(
+            _moderator != address(0),
+            "ThriveProtocol: moderator address is required"
+        );
+        require(
+            _badgeQuery != address(0),
+            "ThriveProtocol: badgeQuery address is required"
+        );
+        require(
+            _deadline > block.timestamp,
+            "ThriveProtocol: deadline must be in the future"
+        );
+
+        moderator = _moderator;
+        rewardToken = _rewardToken;
+        rewardAmount = _rewardAmount;
+        maxRewards = _maxRewards;
+        deadline = _deadline;
+        maxCompletionsPerUser = _maxCompletionsPerUser;
+
+        for (uint256 i = 0; i < _validators.length; i++) {
+            validators.add(_validators[i]);
+        }
+
+        validationMetadata = _validationMetadata;
+        metadataVersion = _metadataVersion;
+        metadata = _metadata;
+        badgeQuery = IBadgeQuery(_badgeQuery);
+    }
+
+    function initialize() external payable onlyModerator {
+        require(!ready, "ThriveProtocol: already initialized");
+        require(
+            validationRewardAmount > 0,
+            "ThriveProtocol: validation reward amount not set"
+        );
```

Comment:
this should be as part of constructor, and same check should be for rewardAmount